cli/cmd/inject.go

```diff
@@ -237,6 +236,7 @@ func injectPodSpec(t *v1.PodSpec, identity k8s.TLSIdentity, controlPlaneDNSNameO
 	if options.disableExternalProfiles {
 		profileSuffixes = "svc.cluster.local."
 	}
+	identity.Namespace = fmt.Sprintf("$(%s)", PodNamespaceEnvVarName)
```

Comment:
why not just set this when the `identity` struct is instantiated, prior to calling `injectPodSpec`?

Reply:
the intent is that this is where the environment is actually instrumented, so this should  be where that templating logic belongs. if it's encoded where the identity struct is instantiated, it seems like a random implementation detail that we have to assume is correct.

More generally, that identity struct is going to be removed/changed substantially over the next few PRs.